src/util/amount.rs

```diff
@@ -129,6 +129,11 @@ impl error::Error for ParseAmountError {
     }
 }
 
+
+fn is_too_precise(s: &str, precision: usize) -> bool {
```

Comment:
Please add a comment here describing this check

Reply:
This could even be inlined again IMO, but if taken out as a method, please provide some docs for context. Implementation seems correct.